mongo/gridfs/bucket.go

```diff
@@ -455,6 +452,47 @@ func (b *Bucket) findChunks(ctx context.Context, fileID interface{}) (*mongo.Cur
 	return chunksCursor, nil
 }
 
+// returns true if the 2 index documents are equal
+func indexDocsEqual(expected, actual bsoncore.Document) (bool, error) {
+	if bytes.Equal(expected, actual) {
+		return true, nil
+	}
+
+	keyElemElems, err := actual.Elements()
+	if err != nil {
+		return false, err
+	}
+	modelKeysElems, err := expected.Elements()
+	if err != nil {
+		return false, err
+	}
+
+	if len(keyElemElems) != len(modelKeysElems) {
+		return false, nil
+	}
+
+	for _, elem := range keyElemElems {
```

Comment:
This doesn't use `modelKeysElems`. Ordering matters for index specs, so we can change this loop to use a counter instead of a range followed by a `LookupErr`:

```
for idx, expectedElem := range expectedElems {
   actualElem := actualElems[idx]
}
```

Reply:
done